_posts/2016-07-07-Rust-1.10.md

```diff
@@ -0,0 +1,270 @@
+---
+layout: post
+title: "Announcing Rust 1.10"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.10. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+As always, you can [install Rust 1.10][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.10][notes] on GitHub.
+1276 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1100-2016-07-07
+
+### What's in 1.10 stable
+
+Rust 1.10 contains one of the most-requested features in Rust: through the [`-C
+panic=abort` flag] or a [setting in your `Cargo.toml`], when a `panic!` happens,
+it will abort, rather than unwind. Why would you want to do this? With an
+abort, less code gets generated, meaning that binary sizes are a bit smaller,
+and compilation time is ever-so-slightly faster. You give up the ability to
+recover, however.
+
+[`-C panic=abort` flag]: https://github.com/rust-lang/rust/pull/32900
+[setting in your `Cargo.toml`]: https://github.com/rust-lang/cargo/pull/2687
+[RFC 1513]: https://github.com/rust-lang/rfcs/blob/master/text/1513-less-unwinding.md
+
+The second big feature in 1.10 is a new crate type: [`cydlib`]. Rust already has
+one dynamic library format, `dylib`. These libraries are currently used everywhere
+you need a dynamic library written in Rust, but as it turns out, there are two major
+use-cases for them: writing a dynamic library to be used in a Rust project, and writing
+a dynamic library to be embedded in some other language. The `dylib` format is focused
+on the first case, and contains extraneous information that's unimportant for the second,
+such as Rust-specific metadata. One final note about this feature: support is in the Rust
+compiler, but not yet in Cargo.
+
+[`cdylib`]: https://github.com/rust-lang/rust/pull/33553
+[RFC 1510]: https://github.com/rust-lang/rfcs/blob/master/text/1510-rdylib.md
+
+[RFC 1521] has landed. This RFC ensures that every type that is `Copy` also [has
+a `Clone` implementation]. Given that `Copy` is an implicit copy, `Clone` is an explicit
+copy, the fact that some types which were `Copy` but not `Clone` felt wrong: if you can
+implicitly copy something, you should be able to also call it explicitly. (But not the
+reverse: many types are `Clone` but not `Copy`.) This has now been rectified.
```

Comment:
This seems very wrong. `Copy: Clone` has been true for a very long time. RFC 1521 merely states that code is allowed to assume that the `Clone` impl for a type is trivial if it is also `Copy`. This basically allows code to specialize on `Copy` types and replace `Clone` calls with simple copies without any change in behavior. There are still types which are `Copy` but not `Clone`, such as fixed size arrays. That has not been fixed and is still a huge hole in Rust's type system.